### PR TITLE
KAFKA-10770: Remove duplicate defination of Metrics#getTags

### PR DESCRIPTION
**Pull request by ArunParthiban-ST**
*Removal of duplicate code.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/metrics/internals/MetricsUtils.java [hunk 1]
```diff
@@ -43,4 +47,20 @@ public static double convert(long timeMs, TimeUnit unit) {
...
+    /**
+     * Create a set of tags using the supplied key and value pairs. The order of the tags will be kept.
+     *
+     * @param keyValue the key and value pairs for the tags; must be an even number
+     * @return the map of tags that can be supplied to the {@link Metrics} methods; never null
+     */
+    public static Map<String, String> getTags(String... keyValue) {
+        if ((keyValue.length % 2) != 0)
+            throw new IllegalArgumentException("keyValue needs to be specified in pairs");
+        Map<String, String> tags = new LinkedHashMap<>();
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/metrics/internals/MetricsUtils.java]: Could you set the initial capability?

File: clients/src/test/java/org/apache/kafka/common/metrics/internals/MetricsUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    public void testCreatingTags() {
+        Map<String, String> tags = MetricsUtils.getTags("k1", "v1", "k2", "v2");
+        assertEquals("v1", tags.get("k1"));
+        assertEquals("v2", tags.get("k2"));
+        assertEquals(2, tags.size());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testCreatingTagsWithOddNumberOfTags() {
+        MetricsUtils.getTags("k1", "v1", "k2", "v2", "extra");
```
Comment (chia7712) [inline clients/src/test/java/org/apache/kafka/common/metrics/internals/MetricsUtilsTest.java]: Could you use ```assertThrows```?

**Review by chia7712 — COMMENTED** [review]
@arunmahadevan Thanks for your patch. some minor comments are left. Please take a look :)

File: clients/src/test/java/org/apache/kafka/common/metrics/internals/MetricsUtilsTest.java (hunk 2 above)
Comment (ArunParthiban-ST) [reply clients/src/test/java/org/apache/kafka/common/metrics/internals/MetricsUtilsTest.java]: Done

File: clients/src/main/java/org/apache/kafka/common/metrics/internals/MetricsUtils.java (hunk 1 above)
Comment (ArunParthiban-ST) [reply clients/src/main/java/org/apache/kafka/common/metrics/internals/MetricsUtils.java]: Sure!

**Comment by ArunParthiban-ST** [comment]
@chia7712   Updated PR. Please take a look.

**Review by chia7712 — APPROVED** [review]
@ArunParthiban-ST Thanks for your patch. LGTM

I will merge it after this weekend (to wait for more reviews)

**Pull request merged**